.github/workflows/tests.yml

```diff
@@ -104,7 +80,7 @@ jobs:
       - name: "Install dependencies with Composer"
         uses: "ramsey/composer-install@3.0.0"
         with:
-          composer-options: "--no-suggest"
+          composer-options: "--no-suggest ${{ matrix.php-version == '8.4' && '--ignore-platform-req=php+' || '' }}"
```

Comment:
TIL about [ternary syntax](https://docs.github.com/en/actions/learn-github-actions/expressions#example) in GitHub Action expressions.

Also, [`--ignore-platform-req=php+`](https://getcomposer.org/doc/03-cli.md#install-i).

What necessitates removing the upper bound for the PHP version? PHPLIB itself allows `^8.0`, so is this for some dev dependency?

Reply:
[psalm](https://github.com/vimeo/psalm/blob/5.x/composer.json#L18) have strict PHP constraint. I added a comment.